src/noise/noise_model.hpp

```diff
@@ -260,6 +260,10 @@ NoiseModel::param_gate_table_ = {
 Circuit NoiseModel::sample_noise(const Circuit &circ,
                                  RngEngine &rng,
                                  const Method method) const {
+  // Check edge case of empty circuit
+  if (circ.ops.empty()) {
+    return circ;
+  }
```

Comment:
If `Circuit::set_params` is doing its job correctly, then this shouldn't be necessary, right?  The auto loop over `circ.ops` further down should just operate zero times.

Reply:
I pushed another fix for this and tested it worked for the previous segfault error by commenting out the early return statement.